File: Private/LeetcodeJS/771.JewelsAndStones.js
// numJewelsInStones = (J, S) => {
//   let hash = {};
//   let count = 0;
//   for (let each of S) {
//     hash[each] = (hash[each]) ? hash[each]+= 1 : 1;
//   }
//   for (let letter of J) {
//     hash[letter] ? count += hash[letter] : null;
//   }
//   return count;
// };

numJewelsInStones = (J, S) => {
  // replaces any stones that are not jewels with '' then returns length
  return S.replace(new RegExp(`[^${J}]`, 'g'), '').length;
}

console.log(numJewelsInStones("aA", "aAAbbbb"));
console.log(numJewelsInStones("z", "ZZ"));

File: PracticeProblems/Leetcode/771.JewelsAndStones.py
def numJewelsInStones(J, S):
    count = 0
    for letter in S:
        if letter in J:
            count += 1
    return count

print(numJewelsInStones("aA", "aAAbbbb"))
print(numJewelsInStones("z", "ZZ"))
